src/utils/hooks/useLocalStorage.tsx

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useState } from 'react';
+
+export const useLocalStorage = (key: string, initialValue: any) => {
+  const getValues = () => {
+    const storage = localStorage.getItem(key);
+
+    if (!storage) {
+      localStorage.setItem(key, JSON.stringify(initialValue));
+
+      return initialValue;
+    }
+
+    return JSON.parse(storage || '');
+  };
+
+  const [value, setValue] = useState<any>(getValues());
+
+  const setTodos = (newValue: any) => {
```

Comment:
The function name 'setTodos' might not be appropriate, given that this is a hook for general use with local storage and not just for todos. Consider using a more generic function name, like 'setStoredValue'.